mvn install does not work

Running `mvn install` causes a build failure.

```
~/p/Scientist4J ❯❯❯ mvn clean install
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 2 modules...
---------------------------------------------------
constituent[0]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/aether-api-1.0.2.v20150114.jar
constituent[1]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/aether-connector-basic-1.0.2.v20150114.jar
constituent[2]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/aether-impl-1.0.2.v20150114.jar
constituent[3]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/aether-spi-1.0.2.v20150114.jar
constituent[4]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/aether-transport-wagon-1.0.2.v20150114.jar
constituent[5]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/aether-util-1.0.2.v20150114.jar
constituent[6]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/aopalliance-1.0.jar
constituent[7]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/cdi-api-1.0.jar
constituent[8]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/commons-cli-1.2.jar
constituent[9]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/commons-io-2.2.jar
constituent[10]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/commons-lang-2.6.jar
constituent[11]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/commons-lang3-3.4.jar
constituent[12]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/guava-18.0.jar
constituent[13]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/guice-4.0-no_aop.jar
constituent[14]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/javax.inject-1.jar
constituent[15]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/jsoup-1.7.2.jar
constituent[16]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/jsr250-api-1.0.jar
constituent[17]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/maven-aether-provider-3.3.9.jar
constituent[18]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/maven-artifact-3.3.9.jar
constituent[19]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/maven-builder-support-3.3.9.jar
constituent[20]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/maven-compat-3.3.9.jar
constituent[21]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/maven-core-3.3.9.jar
constituent[22]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/maven-embedder-3.3.9.jar
constituent[23]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/maven-model-3.3.9.jar
constituent[24]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/maven-model-builder-3.3.9.jar
constituent[25]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/maven-plugin-api-3.3.9.jar
constituent[26]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/maven-repository-metadata-3.3.9.jar
constituent[27]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/maven-settings-3.3.9.jar
constituent[28]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/maven-settings-builder-3.3.9.jar
constituent[29]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/org.eclipse.sisu.inject-0.3.2.jar
constituent[30]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/org.eclipse.sisu.plexus-0.3.2.jar
constituent[31]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/plexus-cipher-1.7.jar
constituent[32]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/plexus-component-annotations-1.6.jar
constituent[33]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/plexus-interpolation-1.21.jar
constituent[34]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/plexus-sec-dispatcher-1.3.jar
constituent[35]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/plexus-utils-3.0.22.jar
constituent[36]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/slf4j-api-1.7.5.jar
constituent[37]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/slf4j-simple-1.7.5.jar
constituent[38]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/wagon-file-2.10.jar
constituent[39]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/wagon-http-2.10-shaded.jar
constituent[40]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/wagon-http-shared-2.10.jar
constituent[41]: file:/usr/local/Cellar/maven/3.3.9/libexec/lib/wagon-provider-api-2.10.jar
constituent[42]: file:/usr/local/Cellar/maven/3.3.9/libexec/conf/logging/
---------------------------------------------------
Exception in thread "main" java.lang.NoClassDefFoundError: com/google/common/base/Strings
	at org.sonatype.nexus.maven.staging.deploy.DeployLifecycleParticipant.getPluginByGAFromContainer(DeployLifecycleParticipant.java:239)
	at org.sonatype.nexus.maven.staging.deploy.DeployLifecycleParticipant.getNexusMavenPluginFromContainer(DeployLifecycleParticipant.java:183)
	at org.sonatype.nexus.maven.staging.deploy.DeployLifecycleParticipant.getBuildPluginsNexusMavenPlugin(DeployLifecycleParticipant.java:164)
	at org.sonatype.nexus.maven.staging.deploy.DeployLifecycleParticipant.afterProjectsRead(DeployLifecycleParticipant.java:96)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:266)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: java.lang.ClassNotFoundException: com.google.common.base.Strings
	at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:50)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass(ClassRealm.java:271)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:247)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:239)
	... 18 more
```

Commenting out `nexus-staging-maven-plugin` causes the build to pass (but causes loss of functionality?). Is there setup that's required to be able to install this artifact locally?

--- a/Scientist4JCore/pom.xml
+++ b/Scientist4JCore/pom.xml
@@ -28,6 +28,11 @@
             <version>3.0.0</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <version>2.5.7</version>
+        </dependency>
     </dependencies>
 
 </project>

--- a/Scientist4JCore/src/main/java/com/github/rawls238/scientist4j/Experiment.java
+++ b/Scientist4JCore/src/main/java/com/github/rawls238/scientist4j/Experiment.java
@@ -12,6 +12,7 @@ import java.util.concurrent.ExecutionException;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
+import java.util.function.BiFunction;
 import java.util.function.Supplier;
 
 public class Experiment<T> {
@@ -27,6 +28,7 @@ public class Experiment<T> {
     private final Counter mismatchCount;
     private final Counter candidateExceptionCount;
     private final Counter totalCount;
+    private final BiFunction<T, T, Boolean> comparator;
 
     public Experiment() {
         this("Experiment");
@@ -57,9 +59,15 @@ public class Experiment<T> {
     }
 
     public Experiment(String name, Map<String, Object> context, boolean raiseOnMismatch, MetricRegistry metricRegistry) {
+        this(name, context, raiseOnMismatch, metricRegistry, Object::equals);
+    }
+
+    public Experiment(String name, Map<String, Object> context, boolean raiseOnMismatch, MetricRegistry metricRegistry, BiFunction<T, T, Boolean> comparator) {
         this.name = name;
         this.context = context;
         this.raiseOnMismatch = raiseOnMismatch;
+        this.comparator = comparator;
+
         controlTimer = getMetrics(metricRegistry).timer(MetricRegistry.name(NAMESPACE_PREFIX, this.name, "control"));
         candidateTimer = getMetrics(metricRegistry).timer(MetricRegistry.name(NAMESPACE_PREFIX, this.name, "candidate"));
         mismatchCount = getMetrics(metricRegistry).counter(MetricRegistry.name(NAMESPACE_PREFIX, this.name, "mismatch"));
@@ -168,7 +176,7 @@ public class Experiment<T> {
     }
 
     protected boolean compareResults(T controlVal, T candidateVal) {
-        return controlVal.equals(candidateVal);
+        return comparator.apply(controlVal, candidateVal);
     }
 
     public boolean compare(Observation<T> controlVal, Observation<T> candidateVal) throws MismatchException {

--- a/Scientist4JCore/src/main/java/com/github/rawls238/scientist4j/Experiment.java
+++ b/Scientist4JCore/src/main/java/com/github/rawls238/scientist4j/Experiment.java
@@ -67,7 +67,6 @@ public class Experiment<T> {
         this.context = context;
         this.raiseOnMismatch = raiseOnMismatch;
         this.comparator = comparator;
-
         controlTimer = getMetrics(metricRegistry).timer(MetricRegistry.name(NAMESPACE_PREFIX, this.name, "control"));
         candidateTimer = getMetrics(metricRegistry).timer(MetricRegistry.name(NAMESPACE_PREFIX, this.name, "candidate"));
         mismatchCount = getMetrics(metricRegistry).counter(MetricRegistry.name(NAMESPACE_PREFIX, this.name, "mismatch"));
@@ -101,7 +100,7 @@ public class Experiment<T> {
 
     private T runSync(Supplier<T> control, Supplier<T> candidate) throws Exception {
         Observation<T> controlObservation;
-        Optional<Observation> candidateObservation = Optional.empty();
+        Optional<Observation<T>> candidateObservation = Optional.empty();
         if (Math.random() < 0.5) {
             controlObservation = executeResult("control", controlTimer, control, true);
             if (runIf() && enabled()) {
@@ -115,13 +114,13 @@ public class Experiment<T> {
         }
 
         countExceptions(candidateObservation, candidateExceptionCount);
-        Result<T> result = new Result(this, controlObservation, candidateObservation, context);
+        Result<T> result = new Result<T>(this, controlObservation, candidateObservation, context);
         publish(result);
         return controlObservation.getValue();
     }
 
     public T runAsync(Supplier<T> control, Supplier<T> candidate) throws Exception {
-        Future<Optional<Observation>> observationFutureCandidate = null;
+        Future<Optional<Observation<T>>> observationFutureCandidate = null;
         Future<Observation<T>> observationFutureControl;
 
         if (Math.random() < 0.5) {
@@ -142,25 +141,25 @@ public class Experiment<T> {
         } catch (InterruptedException | ExecutionException e) {
             throw new RuntimeException(e);
         }
-        Optional<Observation> candidateObservation = Optional.empty();
+        Optional<Observation<T>> candidateObservation = Optional.empty();
         if (observationFutureCandidate != null) {
             candidateObservation = observationFutureCandidate.get();
         }
 
         countExceptions(candidateObservation, candidateExceptionCount);
-        Result<T> result = new Result(this, controlObservation, candidateObservation, context);
+        Result<T> result = new Result<>(this, controlObservation, candidateObservation, context);
         publish(result);
         return controlObservation.getValue();
     }
 
-    private void countExceptions(Optional<Observation> observation, Counter exceptions) {
+    private void countExceptions(Optional<Observation<T>> observation, Counter exceptions) {
         if (observation.isPresent() && observation.get().getException().isPresent()) {
             exceptions.inc();
         }
     }
 
     public Observation<T> executeResult(String name, Timer timer, Supplier<T> control, boolean shouldThrow) throws Exception {
-        Observation<T> observation = new Observation(name, timer);
+        Observation<T> observation = new Observation<>(name, timer);
         observation.startTimer();
         try {
             observation.setValue(control.get());

--- a/Scientist4JCore/src/main/java/com/github/rawls238/scientist4j/ExperimentBuilder.java
+++ b/Scientist4JCore/src/main/java/com/github/rawls238/scientist4j/ExperimentBuilder.java
@@ -1,0 +1,40 @@
+package com.github.rawls238.scientist4j;
+
+import com.codahale.metrics.MetricRegistry;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
+
+public class ExperimentBuilder<T> {
+    private String name;
+    private MetricRegistry registry;
+    private BiFunction<T, T, Boolean> comparator;
+    private Map<String, Object> context;
+
+    public ExperimentBuilder() {
+        context = new HashMap<>();
+        comparator = Object::equals;
+        registry = new MetricRegistry();
+    }
+
+    public ExperimentBuilder<T> withName(final String name) {
+        this.name = name;
+        return this;
+    }
+
+    public ExperimentBuilder<T> withRegistry(final MetricRegistry registry) {
+        this.registry = registry;
+        return this;
+    }
+
+    public ExperimentBuilder<T> withComparator(final BiFunction<T, T, Boolean> comparator) {
+        this.comparator = comparator;
+        return this;
+    }
+
+    public Experiment<T> build() {
+        assert name != null;
+        return new Experiment<>(name, context, false, registry, comparator);
+    }
+}

--- a/Scientist4JCore/src/test/java/com/github/rawls238/scientist4j/ExperimentTest.java
+++ b/Scientist4JCore/src/test/java/com/github/rawls238/scientist4j/ExperimentTest.java
@@ -6,8 +6,10 @@ import com.github.rawls238.scientist4j.exceptions.MismatchException;
 import org.junit.Test;
 
 import java.util.Date;
+import java.util.function.BiFunction;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
 
 public class ExperimentTest {
 
@@ -134,7 +136,17 @@ public class ExperimentTest {
     }
 
     @Test
-    public void shouldUseCustomComparator() {
-        new Experiment<Integer>("test", new MetricRegistry(), )
+    public void shouldUseCustomComparator() throws Exception {
+        @SuppressWarnings("unchecked")
+        final BiFunction<Integer, Integer, Boolean> comparator = mock(BiFunction.class);
+        when(comparator.apply(1, 2)).thenReturn(false);
+        final Experiment<Integer> e = new ExperimentBuilder<Integer>()
+                .withName("test")
+                .withComparator(comparator)
+                .build();
+
+        e.run(() -> 1, () -> 2);
+
+        verify(comparator).apply(1, 2);
     }
 }

--- a/Scientist4JCore/src/test/java/com/github/rawls238/scientist4j/ExperimentTest.java
+++ b/Scientist4JCore/src/test/java/com/github/rawls238/scientist4j/ExperimentTest.java
@@ -48,7 +48,7 @@ public class ExperimentTest {
 
     @Test
     public void itDoesntThrowAnExceptionWhenCandidateFails() {
-        Experiment<Integer> experiment = new Experiment("test");
+        Experiment<Integer> experiment = new Experiment<>("test");
         boolean candidateThrew = false;
         Integer val = 0;
         try {
@@ -64,7 +64,7 @@ public class ExperimentTest {
 
     @Test
     public void itThrowsOnMismatch() {
-        Experiment<Integer> experiment = new Experiment("test", true);
+        Experiment<Integer> experiment = new Experiment<>("test", true);
         boolean candidateThrew = false;
         try {
             experiment.run(this::safeFunction, this::safeFunctionWithDifferentResult);
@@ -79,7 +79,7 @@ public class ExperimentTest {
 
     @Test
     public void itDoesNotThrowOnMatch() {
-        Experiment<Integer> exp = new Experiment("test", true);
+        Experiment<Integer> exp = new Experiment<>("test", true);
         boolean candidateThrew = false;
         Integer val = 0;
         try {
@@ -94,7 +94,7 @@ public class ExperimentTest {
 
     @Test
     public void nonAsyncRunsLongTime() {
-        Experiment<Integer> exp = new Experiment("test", true);
+        Experiment<Integer> exp = new Experiment<>("test", true);
         boolean candidateThrew = false;
         Integer val = 0;
         Date date1 = new Date();
@@ -114,7 +114,7 @@ public class ExperimentTest {
 
     @Test
     public void itWorksWithAnExtendedClass() {
-        Experiment<Integer> exp = new TestPublishExperiment("test");
+        Experiment<Integer> exp = new TestPublishExperiment<>("test");
         try {
             exp.run(this::safeFunction, this::safeFunction);
         } catch (Exception e) {
@@ -125,11 +125,16 @@ public class ExperimentTest {
     @Test
     public void candidateExceptionsAreCounted() throws Exception {
         MetricRegistry metrics = new MetricRegistry();
-        Experiment<Integer> exp = new Experiment("test", metrics);
+        Experiment<Integer> exp = new Experiment<>("test", metrics);
 
         exp.run(() -> { return 1; }, this::exceptionThrowingFunction);
 
         Counter result = metrics.getCounters().get("scientist.test.candidate.exception");
         assertThat(result.getCount()).isEqualTo(1);
+    }
+
+    @Test
+    public void shouldUseCustomComparator() {
+        new Experiment<Integer>("test", new MetricRegistry(), )
     }
 }

--- a/Scientist4JCore/src/test/java/com/github/rawls238/scientist4j/ExperimentTest.java
+++ b/Scientist4JCore/src/test/java/com/github/rawls238/scientist4j/ExperimentTest.java
@@ -14,7 +14,7 @@ import static org.mockito.Mockito.*;
 public class ExperimentTest {
 
     private Integer exceptionThrowingFunction() {
-        throw new RuntimeException("throw an exception");
+        throw new ExpectingAnException("throw an exception");
     }
 
     private Integer safeFunction() {
@@ -34,94 +34,49 @@ public class ExperimentTest {
         return 4;
     }
 
-    @Test
-    public void itThrowsAnExceptionWhenControlFails() {
-        Experiment experiment = new Experiment("test");
-        boolean controlThrew = false;
-        try {
-            experiment.run(this::exceptionThrowingFunction, this::exceptionThrowingFunction);
-        } catch (RuntimeException e) {
-            controlThrew = true;
-        } catch (Exception e) {
-
-        }
-        assertThat(controlThrew).isEqualTo(true);
+    @Test(expected = ExpectingAnException.class)
+    public void itThrowsAnExceptionWhenControlFails() throws Exception {
+        new Experiment<Integer>("test")
+                .run(this::exceptionThrowingFunction, this::exceptionThrowingFunction);
     }
 
     @Test
-    public void itDoesntThrowAnExceptionWhenCandidateFails() {
+    public void itDoesntThrowAnExceptionWhenCandidateFails() throws Exception {
         Experiment<Integer> experiment = new Experiment<>("test");
-        boolean candidateThrew = false;
-        Integer val = 0;
-        try {
-            val = experiment.run(this::safeFunction, this::exceptionThrowingFunction);
-        } catch (RuntimeException e) {
-            candidateThrew = true;
-        } catch (Exception e) {
+        Integer val = experiment.run(this::safeFunction, this::exceptionThrowingFunction);
+        assertThat(val).isEqualTo(3);
+    }
 
-        }
-        assertThat(candidateThrew).isEqualTo(false);
+    @Test(expected = MismatchException.class)
+    public void itThrowsOnMismatch() throws Exception {
+        new Experiment<Integer>("test", true)
+                .run(this::safeFunction, this::safeFunctionWithDifferentResult);
+    }
+
+    @Test
+    public void itDoesNotThrowOnMatch() throws Exception {
+        Integer val = new Experiment<Integer>("test", true)
+                .run(this::safeFunction, this::safeFunction);
+
         assertThat(val).isEqualTo(3);
     }
 
     @Test
-    public void itThrowsOnMismatch() {
-        Experiment<Integer> experiment = new Experiment<>("test", true);
-        boolean candidateThrew = false;
-        try {
-            experiment.run(this::safeFunction, this::safeFunctionWithDifferentResult);
-        } catch (MismatchException e) {
-            candidateThrew = true;
-        } catch (Exception e) {
-
-        }
-
-        assertThat(candidateThrew).isEqualTo(true);
-    }
-
-    @Test
-    public void itDoesNotThrowOnMatch() {
+    public void nonAsyncRunsLongTime() throws Exception {
         Experiment<Integer> exp = new Experiment<>("test", true);
-        boolean candidateThrew = false;
-        Integer val = 0;
-        try {
-            val = exp.run(this::safeFunction, this::safeFunction);
-        } catch (Exception e) {
-            candidateThrew = true;
-        }
-
-        assertThat(val).isEqualTo(3);
-        assertThat(candidateThrew).isEqualTo(false);
-    }
-
-    @Test
-    public void nonAsyncRunsLongTime() {
-        Experiment<Integer> exp = new Experiment<>("test", true);
-        boolean candidateThrew = false;
-        Integer val = 0;
         Date date1 = new Date();
-
-        try {
-            val = exp.run(this::sleepFunction, this::sleepFunction);
-        } catch (Exception e) {
-            candidateThrew = true;
-        }
+        Integer val = exp.run(this::sleepFunction, this::sleepFunction);
         Date date2 = new Date();
         long difference = date2.getTime() - date1.getTime();
 
         assertThat(difference).isGreaterThanOrEqualTo(2000);
         assertThat(val).isEqualTo(3);
-        assertThat(candidateThrew).isEqualTo(false);
     }
 
     @Test
-    public void itWorksWithAnExtendedClass() {
+    public void itWorksWithAnExtendedClass() throws Exception {
         Experiment<Integer> exp = new TestPublishExperiment<>("test");
-        try {
-            exp.run(this::safeFunction, this::safeFunction);
-        } catch (Exception e) {
-
-        }
+        exp.run(this::safeFunction, this::safeFunction);
     }
 
     @Test
@@ -129,7 +84,7 @@ public class ExperimentTest {
         MetricRegistry metrics = new MetricRegistry();
         Experiment<Integer> exp = new Experiment<>("test", metrics);
 
-        exp.run(() -> { return 1; }, this::exceptionThrowingFunction);
+        exp.run(() -> 1, this::exceptionThrowingFunction);
 
         Counter result = metrics.getCounters().get("scientist.test.candidate.exception");
         assertThat(result.getCount()).isEqualTo(1);
@@ -148,5 +103,11 @@ public class ExperimentTest {
         e.run(() -> 1, () -> 2);
 
         verify(comparator).apply(1, 2);
+    }
+}
+
+class ExpectingAnException extends RuntimeException {
+    ExpectingAnException(final String message) {
+        super(message);
     }
 }

--- a/pom.xml
+++ b/pom.xml
@@ -29,17 +29,17 @@
     </distributionManagement>
     <build>
         <plugins>
-            <plugin>
-                <groupId>org.sonatype.plugins</groupId>
-                <artifactId>nexus-staging-maven-plugin</artifactId>
-                <version>1.6.3</version>
-                <extensions>true</extensions>
-                <configuration>
-                    <serverId>ossrh</serverId>
-                    <nexusUrl>https://oss.sonatype.org/</nexusUrl>
-                    <autoReleaseAfterClose>true</autoReleaseAfterClose>
-                </configuration>
-            </plugin>
+            <!--<plugin>-->
+                <!--<groupId>org.sonatype.plugins</groupId>-->
+                <!--<artifactId>nexus-staging-maven-plugin</artifactId>-->
+                <!--<version>1.6.3</version>-->
+                <!--<extensions>true</extensions>-->
+                <!--<configuration>-->
+                    <!--<serverId>ossrh</serverId>-->
+                    <!--<nexusUrl>https://oss.sonatype.org/</nexusUrl>-->
+                    <!--<autoReleaseAfterClose>true</autoReleaseAfterClose>-->
+                <!--</configuration>-->
+            <!--</plugin>-->
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-gpg-plugin</artifactId>

--- a/pom.xml
+++ b/pom.xml
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
     <groupId>com.github.rawls238</groupId>
@@ -54,24 +55,9 @@
                 </executions>
             </plugin>
             <plugin>
-                <groupId>org.sonatype.plugins</groupId>
-                <artifactId>nexus-staging-maven-plugin</artifactId>
-                <version>1.6.3</version>
-                <extensions>true</extensions>
-                <configuration>
-                    <serverId>ossrh</serverId>
-                    <nexusUrl>https://oss.sonatype.org/</nexusUrl>
-                    <autoReleaseAfterClose>true</autoReleaseAfterClose>
-                </configuration>
-            </plugin>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-javadoc-plugin</artifactId>
-                <version>2.9.1</version>
-            </plugin>
-            <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-source-plugin</artifactId>
+                <version>3.0.1</version>
                 <executions>
                     <execution>
                         <id>attach-sources</id>
@@ -84,6 +70,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
+                <version>2.9.1</version>
                 <executions>
                     <execution>
                         <id>attach-javadocs</id>
@@ -93,12 +80,8 @@
                     </execution>
                 </executions>
             </plugin>
-
-
-
         </plugins>
     </build>
-
 
     <licenses>
         <license>
@@ -106,7 +89,6 @@
             <url>http://www.opensource.org/licenses/mit-license.php</url>
         </license>
     </licenses>
-
 
     <developers>
         <developer>

